File: erlang/anagram/anagram.erl
-module(anagram).
-export([find/2]).

find(Word, Anagrams) -> 
    lists:filter(anagram_of(Word), Anagrams).

anagram_of(Word) -> 
    fun (Anagram) ->
            string:to_lower(Word) =/= string:to_lower(Anagram) andalso
            string:equal(lists:sort(string:to_lower(Word)),
                         lists:sort(string:to_lower(Anagram)))
    end.

File: clojure/anagram/src/anagram.clj
(ns anagram
  (:require [clojure.string :as s]))

(defn- anagram?
  [word candidate]
  (let [w (s/lower-case word)
        c (s/lower-case candidate)]
    (and (not= w c)
         (= (sort w) (sort c)))))

(defn anagrams-for
  [word candidates]
    (filter (partial anagram? word) candidates))
